OpenSSL/crypto.py

```diff
@@ -25,13 +25,43 @@
 TYPE_DSA = _lib.EVP_PKEY_DSA
 
 
+
 class Error(Exception):
     """
     An error occurred in an `OpenSSL.crypto` API.
     """
 
 
+
+def _exception_from_context_error(exception_type, store_ctx):
+    """
+    Convert a :py:func:`OpenSSL.crypto.verify_cert` failure into a Python
+    exception.
+
+    When a call to native OpenSSL X509_verify_cert fails, additonal information
+    about the failure can be contained from the store context.
+    """
+
+    errors = [
+        _lib.X509_STORE_CTX_get_error(store_ctx._store_ctx),
+        _lib.X509_STORE_CTX_get_error_depth(store_ctx._store_ctx),
+        _native(_ffi.string(_lib.X509_verify_cert_error_string(_lib.X509_STORE_CTX_get_error(store_ctx._store_ctx)))),
+    ]
+    _x509 = _lib.X509_STORE_CTX_get_current_cert(store_ctx._store_ctx)
+    if _x509 != _ffi.NULL:
```

Comment:
`pycert` doesn't get initialized outside of this `if` block.  So if `X509_STORE_CTX_get_current_cert` ever returns NULL then this code will raise a `NameError` a few lines below.

Can this return NULL if the certificate has been set to a non-NULL value?  If so, a test exercising that case would be great.  If not ... maybe the `if` check here is not necessary.  In C that might lead to a NULL dereference (and a possible security issue) if we ever get surprised by a NULL return value.  In Python it just means we'll get an `AttributeError` or a `TypeError` or something on one of the following lines.  For a case we thought was impossible, that's not a terrible failure mode.